python/pyarrow/tests/test_pandas.py

```diff
@@ -3353,6 +3343,13 @@ def test_convert_to_extension_array(monkeypatch):
     assert isinstance(result._data.blocks[0], _int.ExtensionBlock)
     tm.assert_frame_equal(result, df2)
 
+    # # monkeypatch pandas Int64Dtype to *not* have the protocol method
+    # monkeypatch.delattr(pd.Int64Dtype, "__from_arrow__")
```

Comment:
Still looking into how to get this working (it seems a problem on the pandas side), as it would be nice to still have a test for the case that a pandas ExtensionDtype does not have this `__from_arrow__` method